emu/src/cpu/single_data_transfer.rs

```diff
@@ -113,131 +104,232 @@ impl Arm7tdmi {
 
 #[cfg(test)]
 mod tests {
```

Comment:
Mhh, I don't think we should change all these tests like this. We're losing tests over the functionalities